druid-shell/src/platform/x11/window.rs

```diff
@@ -602,6 +602,42 @@ impl Window {
         }
     }
 
+    pub fn handle_wheel(&self, event: &xcb::ButtonPressEvent) -> Result<(), Error> {
+        let button = event.detail();
+        // The scroll delta of 120.0 was stolen from the GTK code, which apparently came from the
+        // windows docs.
+        let delta = match button {
+            4 => (0.0, -120.0),
+            5 => (0.0, 120.0),
+            6 => (-120.0, 0.0),
+            7 => (120.0, 0.0),
+            _ => {
+                log::error!("unexpected mouse wheel button: {}", button);
```

Comment:
This is a good spot to return an error instead. Perhaps something like:
```rust
_ => return Err(Error::Generic(format!("unexpected mouse wheel button: {}", button))),
```

Reply:
I'm not sure if this affects your preferred error handling technique, but hitting this case represents a bug in druid-shell. It's impossible for now because we check that the button is between 4 and 7 before calling `handle_wheel`.

Reply:
Yeah I think returning an error in case of invalid input parameters is just fine.